content/blog/lpython_mvp.md

```diff
@@ -62,7 +146,448 @@ All optimizations are applied via one command-line argument, `--fast`. To select
+                                    []
+                                    [(=
+                                        (Var 3 x)
+                                        (IntegerConstant 0 (Integer 4))
+                                        ()
+                                    )
+                                    (=
+                                        (Var 3 i)
+                                        (IntegerBinOp
+                                            (IntegerConstant 0 (Integer 4))
+                                            Sub
+                                            (IntegerConstant 1 (Integer 4))
+                                            (Integer 4)
+                                            ()
+                                        )
+                                        ()
+                                    )
+                                    (WhileLoop
+                                        ()
+                                        (IntegerCompare
+                                            (IntegerBinOp
+                                                (Var 3 i)
+                                                Add
+                                                (IntegerConstant 1 (Integer 4))
+                                                (Integer 4)
+                                                ()
+                                            )
+                                            LtE
+                                            (IntegerConstant 31 (Integer 4))
+                                            (Logical 4)
+                                            ()
+                                        )
+                                        [(=
+                                            (Var 3 i)
+                                            (IntegerBinOp
+                                                (Var 3 i)
+                                                Add
+                                                (IntegerConstant 1 (Integer 4))
+                                                (Integer 4)
+                                                ()
+                                            )
+                                            ()
+                                        )
+                                        (=
+                                            (Var 3 x)
+                                            (IntegerBinOp
+                                                (Var 3 x)
+                                                Add
+                                                (Var 3 i)
+                                                (Integer 4)
+                                                ()
+                                            )
+                                            ()
+                                        )
+                                        ... # Above two assignments are repeated 30 times because of loop_unroll
```

Comment:
Does this make sense?
Both ASR are 177 and 240 lines

Reply:
You can show the example of inline function calls? That should result in smaller ASR.

Reply:
Yup! I think that should be small.

Reply:
I get almost similar lines: 190 lines && 210 lines for the following program. I was not able to simplify further.
```
from lpython import i32

def compute_x() -> i32:
    return (2 * 3) ** 1 + 2

def main():
    x: i32 = compute_x()
    print(x)

main()
```

Reply:
We can try unused_function, dead_code_removal